main.go

```diff
@@ -119,17 +114,59 @@ func main() {
 
 			// output identifier accessing methods, checking that we support the field type
 			for _, ident := range idents {
-				switch ident.Identifier.Desc.Kind() {
-				case protoreflect.StringKind:
-					output.P(fmt.Sprintf(tmpl, ident.Msg.Desc.Name(), ident.Identifier.GoName))
-				case protoreflect.EnumKind:
-					output.P(fmt.Sprintf(tmpl, ident.Msg.Desc.Name(), ident.Identifier.GoName+".String()"))
-				default:
-					return fmt.Errorf("unsupported field type on %s: %s", ident.Msg.Desc.Name(), ident.Identifier.Desc.Kind())
-				}
+				output.P(fmt.Sprintf(tmpl, ident.message, ident.identifier))
 			}
 		}
 
 		return nil
 	})
 }
+
+func getMessageIdentifier(entity string, msg *protogen.Message, fields ...string) (identifier, error) {
+	for _, field := range msg.Fields {
+		// Skip field that don't have `uw.entity.v1.identifer = true`
+		fieldopts := field.Desc.Options().(*descriptorpb.FieldOptions)
+
+		if !proto.GetExtension(fieldopts, entitypb.E_Identifier).(bool) {
+			continue
+		}
+
+		if field.Desc.Cardinality() == protoreflect.Repeated {
+			return identifier{}, fmt.Errorf("repeated fields are not supported on %s: %s: %w", entity, field.Desc.Kind(), errUnsupportedField)
+		}
+
+		var value string
+		switch field.Desc.Kind() {
+		case protoreflect.StringKind:
+			value = fmt.Sprintf("Get%s()", field.GoName)
+		case protoreflect.EnumKind:
+			value = fmt.Sprintf("Get%s().String()", field.GoName)
+		case protoreflect.MessageKind:
+			nestedIdentifier := false
+			// check if the nested message has an identifier
+			for _, f := range field.Message.Fields {
+				if proto.GetExtension(f.Desc.Options().(*descriptorpb.FieldOptions), entitypb.E_Identifier).(bool) {
+					nestedIdentifier = true
+				}
+			}
+
+			if !nestedIdentifier {
+				return identifier{}, fmt.Errorf("nested messages must have an identifier on %s: %s: %w", entity, field.Desc.Kind(), errUnsupportedField)
+			}
```

Comment:
Think it's worth adding a test for this.

Reply:
I'll try to think about how to add one for it. For now I've added screenshot explaining how that error can happen.

Reply:
hah, didn't realize it's not straightforward to programatically test